platform/features/timeline/src/controllers/swimlane/TimelineSwimlaneDropHandler.js

```diff
@@ -129,17 +155,12 @@ define(
                 });
             }
 
-            // Check if a compose action is allowed for the object in this
-            // swimlane (we handle the link differently to set the index,
-            // but check for the existence of the action to invole the
-            // relevant policies.)
-            function allowsCompose(swimlane, domainObject) {
-                var actionCapability =
-                    swimlane.domainObject.getCapability('action');
-                return actionCapability && actionCapability.getActions({
-                    key: 'compose',
-                    selectedObject: domainObject
-                }).length > 0;
+            function drop(domainObject, targetObject, indexOffset) {
+                var action = chooseAction(targetObject, domainObject);
+                return action.perform().then(function () {
```

Comment:
`chooseAction` is also used to determine drop eligibility (is there an action to satisfy this drop with?). Dropping on self to reorder is an unusual case where there is no such action, but we still want to allow it; there are other cases when there is no such action (e.g. dropping a Folder into an Activity) and we do want this to be disallowed

I could rewrite this to relocate the special-case logic - thoughts on what would be easier to follow here?